qiskit/pulse/commands/sample_pulse.py

```diff
@@ -88,13 +88,12 @@ def __eq__(self, other: 'SamplePulse'):
         Returns:
             bool: are self and other equal
         """
-        if super().__eq__(other) and \
-                (self._samples == other._samples).all():
+        if super().__eq__(other) and (self.samples == other.samples).all():
             return True
         return False
 
     def __hash__(self):
-        return hash((super().__hash__(), self._samples.tostring()))
+        return hash((super().__hash__(), self.samples.tostring()))
```

Comment:
where is sample pulse being hashed? out of scope, I know. your change here is good

Reply:
It is not being hashed anywhere right now. However, it is recommended in Python to override `__hash__` for immutable objects. This way `self.__eq__(other)` will always match `self.hash()==other.hash()`. [See for more info](https://docs.python.org/3/reference/datamodel.html#object.__hash__).